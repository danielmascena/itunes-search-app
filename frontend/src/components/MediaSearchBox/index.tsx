import * as React from "react";
import Box from "@mui/material/Box";
import TextField from "@mui/material/TextField";

interface Props {
  onChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
  placeholder?: string
}

const MediaSearchBox: React.FC<Props> = (props) => (
  <Box
    component="form"
    sx={{
      "& .MuiTextField-root": { m: 1, width: "25ch" },
    }}
    noValidate
    autoComplete="off"
  >
    <TextField
      id="search-textbox"
      name="search-textbox"
      inputProps={{ "data-testid": "search-textbox" }}
      label="Search"
      type="search"
      variant="standard"
      {...props}
    />
  </Box>
);

export default MediaSearchBox;
