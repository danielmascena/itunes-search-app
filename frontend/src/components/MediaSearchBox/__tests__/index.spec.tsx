import * as React from "react";
import renderer from "react-test-renderer";
import { render, fireEvent, screen, cleanup } from "@testing-library/react";

import MediaSearchBox from "../index";

const hasInputValue = (element: TestElement, inputValue: string) =>
  screen.getByDisplayValue(inputValue) === element;

const changeHandler = jest.fn();

describe("MediaSearchBox Component", () => {
  afterEach(cleanup);

  it("renders the MediaSearchBox component", () => {
    expect(
      renderer.create(<MediaSearchBox onSearch={changeHandler} />).toJSON()
    ).toMatchSnapshot();
  });

  it("calls 'onSearch' prop on input change", async () => {
    const querySearch = "I am with the band";
    const placeholder = "Artists, songs, or albums";
    render(
      <MediaSearchBox placeholderText={placeholder} onSearch={changeHandler} />
    );
    const input = screen.getByPlaceholderText("Artists, songs, or albums");

    fireEvent.change(input, { target: { value: querySearch } });
    expect(hasInputValue(input, querySearch)).toBe(true);
  });
});
