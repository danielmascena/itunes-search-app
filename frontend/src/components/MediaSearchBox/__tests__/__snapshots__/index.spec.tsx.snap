// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MediaSearchBox Component renders the MediaSearchBox component 1`] = `
<form
  autoComplete="off"
  className="MuiBox-root css-m4cqzj"
  noValidate={true}
>
  <div
    className="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
  >
    <label
      className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink={false}
      htmlFor="search-textbox"
      id="search-textbox-label"
    >
      Search
    </label>
    <div
      className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoFocus={false}
        className="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputTypeSearch css-1g24dm6-MuiInputBase-input-MuiOutlinedInput-input"
        data-testid="search-textbox"
        disabled={false}
        id="search-textbox"
        name="search-textbox"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="Search"
        required={false}
        type="search"
        value=""
      />
      <fieldset
        aria-hidden={true}
        className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          className="css-1ftyaf0"
        >
          <span>
            Search
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</form>
`;
